lib/timeline.js

```diff
@@ -0,0 +1,107 @@
+class Timeline {
+  constructor(client) {
+    this.client = client
+  }
+
+  createEventType(applicationId, userId, data, cb) {
+    return this.client._request(
+      {
+        method: 'POST',
+        path: `/integrations/v1/${applicationId}/timeline/event-types?userId=${userId}`,
+        body: data,
+      },
+      cb,
+    )
+  }
+
+  updateEventType(eventId, applicationId, data, cb) {
+    const parameters = {
+      method: 'PUT',
+      path: `/integrations/v1/${applicationId}/timeline/event-types/${eventId}`,
+      body: data,
+    }
+
+    return this.client._request(parameters, cb)
+  }
+
+  createEventTypeProperty(eventId, applicationId, userId, data, cb) {
+    const parameters = {
+      method: 'POST',
+      path: `/integrations/v1/${applicationId}/timeline/event-types/${eventId}/properties?userId=${userId}`,
+      body: data,
+    }
+
+    return this.client._request(parameters, cb)
+  }
+
+  updateEventTypeProperty(eventId, applicationId, propertyId, data, cb) {
+    const parameters = {
+      method: 'PUT',
+      path: `/integrations/v1/${applicationId}/timeline/event-types/${eventId}/properties/${propertyId}`,
+      body: data,
+    }
+
+    return this.client._request(parameters, cb)
+  }
+
+  createTimelineEvent(applicationId, eventTypeId, data, cb) {
+    if (
+      !this.client.clientId ||
+      !this.client.clientSecret ||
+      !this.client.redirectUri ||
+      !this.client.refreshToken
+    ) {
+      return cb(
+        new Error(
+          'You must init hubspot with a clientId/clientSecret/redirectUri/refreshToken and call hubspot.refreshAccessToken()',
+        ),
+      )
+    }
+
+    if (!data || (!data.email && !data.utk && !data.objectId)) {
```

Comment:
I've tested not passing these and the HubSpot API returns an appropriate error if one of these values are not present. I think it's fine for us not to make this check.